### ISSUE TITLE
Feature Pyramid Network should share RPN weights between layers.

### ISSUE THREAD

**Issue by ChristopherAkroyd** [broadinstitute/keras-rcnn#166]
## Problem and Details

Within the models/_rpn.py code the Feature Pyramid Network is implemented with a separate RPN for each different pyramid level whereas from the paper on [FPN's](https://arxiv.org/pdf/1612.03144.pdf) we can see from section 4 that their implementation shares weights between all pyramid levels. The justification for this was based on the researchers observing a similar level of accuracy when sharing and not sharing, therefore we could make this weight sharing optional.

## Solution

The solution is quite simple, instead of defining a new set of RPN layers for each level we define a shared set of RPN layers outside of the loop and call them using the Keras functional API for each layer.

        shared_rpn_conv_3x3 = keras.layers.Conv2D(features, **options)
        shared_rpn_deltas = keras.layers.Conv2D(number_of_anchors * 4, (1, 1), activation="linear", kernel_initializer="zero")
        shared_rpn_scores = keras.layers.Conv2D(number_of_anchors * 1, (1, 1), activation="sigmoid", kernel_initializer="uniform")

        for index, feature_map in enumerate(feature_maps):
            name = f"p{index + 2}"

            convolution_3x3 = shared_rpn_conv_3x3(locals()[name])
            deltas = shared_rpn_deltas(convolution_3x3)
            scores = shared_rpn_scores(convolution_3x3)
            #remainder of code ommitted

Ultimately this change results in a reduction of 4 million parameters in the model when tested on the malaria dataset at a minor cost to the decrease in the loss rate.

## Further Action
I'll be doing some experiments tonight to quantify the impact of the change tonight but I can have a PR ready to go for tomorrow. Please let me know your thoughts and if there's anything else I can help with.

## Relevant Links

- [Feature Pyramid Networks](https://arxiv.org/pdf/1612.03144.pdf)
